### PR TITLE
Return values: last half mile

### PR DESCRIPTION
**Pull request by mih**
TODO:

- [x] Give summary according to #1476
- [x] converted `add` to a new-style command
- [ ] figure out what to do with the `ValueError` thrown by `Interface._prep()` if non-dataset paths are given. This should rather turn into an "error" result to get a result for each input argument.
- [x] make `add` talk about all non-actions like `get`. For example: currently a `datalad add .` in a clean dataset gives no result whatsoever
- [x] be able to `add` and `save` things that are already staged #1413

### CODE REVIEW

**Comment by mih** [comment]
FTR: This is how it looks now:

Scenario: clean repo, with README.rst in Git

````
% datalad add . .
add(notneeded): /tmp/aligned (directory) [nothing to add from /tmp/aligned]
-> exit 0
% datalad add . README.rst
add(notneeded): /tmp/aligned (directory) [nothing to add from /tmp/aligned]
add(notneeded): /tmp/aligned/README.rst (file) [/tmp/aligned/README.rst is already included in the dataset]
Action summary:
  add (notneeded: 2)
-> exit 0
````

Previously `add` would have been completely silent in both cases.

File: datalad/interface/results.py [hunk 1]
```diff
@@ -246,3 +261,86 @@ def is_result_matching_pathsource_argument(res, **kwargs):
...
+      placeholder that is expanded with the path of that directory.
+    noinfo_file_msg : str
+      Message template to inject into the result for a requested file that `git
+      annex` was silent about. There must be one string placeholder that is
+      expanded with the path of that file.
+    **kwargs
+      Any further kwargs are included in the yielded result dictionary.
+    """
+    while requested_paths:
+        p = requested_paths.pop()
```
Comment (yarikoptic) [inline datalad/interface/results.py]: modifies inplace

Comment (mih) [inline datalad/interface/results.py]: modified.

**Comment by yarikoptic** [comment]
@mih, re 
````
% datalad add .
add(notneeded): /tmp/aligned (directory) [nothing to add from /tmp/aligned]
-> exit 0
````
so how to add all files present in this directory under git/annex?  `-r` is "reserved" somewhat to operate across datasets boundaries, so we have no distinction to say to go under subdirectories...

**Comment by mih** [comment]
This stays the same. In this example there simply were no files that
weren't added yet.

**Comment by mih** [comment]
This PR is a minor addition to the merged #1409, and has a fix that we shouldn't keep from master -> Merging.

**Comment by yarikoptic** [comment]
Now that it was against feature branch and not master

**Comment by yarikoptic** [comment]
Nevermind, find #1499
Thanks!!!

**Pull request merged**